### DISCUSSION TITLE
Add the possibility to add a link description before the link itself.

### DISCUSSION THREAD

**Discussion by anuvyklack** [nvim-neorg/neorg#336] [Ideas]
As I understand the Neorg files should be readable in raw just in Neovim inside the terminal (like markdown).

But links can be quite long, as in the example below, and if the window is not wide enough (which is the case when you work on a laptop and split Neovim window vertically) and link is not on the first level and so indented from the left side, then you can't see the link description at all. For example, the link below is 86 characters long:

``` norg
{https://github.com/nvim-neorg/neorg/blob/main/docs/NFF-0.1-spec.md#the-link-location}[description]
```

If the first curly brace is mandatory for parsing, the description can be placed before link body like this:

``` norg
{[description]link-body}
```

This should improve readability.  The only drawback I can see is that the link can't start with `[` but I don't know if such links exist.

**Comment by max397574** [comment]
check out the concealer settings
there is the option to use the unicode joiner

**Reply by anuvyklack** [reply]
I read concealer module documentation, but don't get how exactly I can do it. Which setting do I need?

**Reply by mrossinek** [reply]
You need the following:
```lua
        ["core.norg.concealer"] = {
            config = {
                markup_preset = "dimmed",
            },
        },
```
`dimmed` is the default which will simply dim concealed chars but still shows them. If you change that to `brave` it will use a unicode word joiner char, effectively replacing all concealed chars with zero-width chars, thereby hiding the concealed text. This has some quirks and can leave behind visual artifacts though which is why this is not the default.

**Reply by anuvyklack** [reply]
I missed wrapping my settings with `config = {...}` table. And `brave` value is not usable at least for me now. But anyway, thank you!

**Comment by mrossinek** [comment]
Thanks for the suggestion! While the concealer settings are a valid approach, this does not solve what I believe you see as the fundamental issue, which is readability of the plain text.
While what you suggest is not possible due to the way we have designed the Norg syntax (with link locations being placed inside `{}` and link descriptions inside `[]`) I recommend that you check out **anchors**. These are designed mostly for use cases where you may want to link to the same location multiple times and want to remove redundancy and save some typing, but they also can be simply used for improved readability. Below is an example for your specific scenario:
```norg
Consider reading the entire [specification] of the Norg syntax.
You can also check out `:help neorg-anchors` for more details!
+Note: I just updated the help tags so be sure to update Neorg in order to find this help page!+

[specification]{https://github.com/nvim-neorg/neorg/blob/main/docs/NFF-0.1-spec.md#the-link-location}
```

**Reply by anuvyklack** [reply]
Yes, you understand me right.
I read about **anchors** but using it out of use case they were designed for is a bit of a crutch.
I think it will be better to have a way to place description before the link.

**Reply by mrossinek** [reply]
Well, essentially anchors are exactly what you are asking for because they allow you to place a description in front of the location. Nothing stops you from writing the following:
```norg
Link to the [specification]{https://github.com/nvim-neorg/neorg/blob/main/docs/NFF-0.1-spec.md#the-link-location}
```
This is still valid norg syntax. But adding the ability to place `[]` inside of `{}` is not something that will happen.

**Reply by anuvyklack** [reply]
No, I don't hold on placing  `[]` inside of `{}` that was just a proposal.

If I can write
```
Link to the [specification]{https://github.com/nvim-neorg/neorg/blob/main/docs/NFF-0.1-spec.md#the-link-location}.
Some other text...
```
without placing anchor definition at the bottom of the document and it is not a mistake, that is exactly what I want!

**Reply by mrossinek** [reply]
Yes. Technically you are still writing anchor definitions but these are not required to be at a certain location in your document. Since this is in-line syntax they can be placed wherever you like, just like normal links :+1:

**Comment by max397574** [comment]
@anuvyklack a lot has changed since this
is this still unanswered?
